File: 04-Secuencial/04-IMC/02-Go/IMC.go
/*
   Programa para calcular el índice de masa corporal (IMC), usando la
   altura y el peso.
   Autor: Julián Esteban Gutiérrez Posada
   Fecha: Marzo 2025
   Licencia: GNU GPL v3
*/

package main

import (
	"fmt"
	"imc/modulo"
	"math"
	"strconv"
)

func main() {
	peso := modulo.IngresarEntero("Peso  (kg): ")
	altura := ingresarReal("Altura (m): ")
	imc := calcularIMC(peso, altura)
	informeIMC := generarInformeIMC(peso, altura, imc)
	modulo.MostrarMensaje(informeIMC)
}

func ingresarReal(pregunta string) float64 {
	real, _ := strconv.ParseFloat(modulo.IngresarTexto(pregunta), 64)
	return real
}

func calcularIMC(peso int, altura float64) float64 {
	return float64(peso) / math.Pow(altura, 2.0)
}

func generarInformeIMC(peso int, altura float64, imc float64) string {
	return fmt.Sprintf(
		"\nCon su peso de %d kg y su altura de %.1f metros"+
			"\nsu índice de masa corporal (IMC) es de %.1f", peso, altura, imc)
}

File: 04-Secuencial/04-IMC/05-Haskell/IMC.hs
{-
   Programa para calcular el índice de masa corporal (IMC), usando la
   altura y el peso.
   Autor: Julián Esteban Gutiérrez Posada
   Fecha: Marzo 2025
   Licencia: GNU GPL v3
-}

import Text.Printf (printf)
import Modulo.Util (mostrarMensaje, ingresarEntero)

main :: IO()
main = do
  peso <- ingresarEntero "Peso  (kg): "
  altura <- ingresarReal "Altura (m): "
  let imc = calcularIMC peso altura
  let informeIMC = generarInformeIMC peso altura imc
  mostrarMensaje informeIMC

ingresarReal :: String -> IO Double
ingresarReal pregunta = do
  mostrarMensaje pregunta
  readLn

calcularIMC :: Int -> Double -> Double
calcularIMC peso altura =
  fromIntegral peso / altura ** 2

generarInformeIMC :: Int -> Double -> Double -> String
generarInformeIMC peso altura imc =
  printf
    ("\nCon su peso de %d kg y su altura de %.1f metros" ++
    "\nsu índice de masa corporal (IMC) es de %.1f")
    peso
    altura
    imc